File: sources/LRQ 1.0/net/ccbluex/liquidbounce/features/module/modules/combat/AutoPot.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  kotlin.TypeCastException
 *  kotlin.jvm.internal.Intrinsics
 */
package net.ccbluex.liquidbounce.features.module.modules.combat;

import kotlin.TypeCastException;
import kotlin.jvm.internal.Intrinsics;
import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.api.enums.WEnumHand;
import net.ccbluex.liquidbounce.api.minecraft.client.entity.IEntityPlayerSP;
import net.ccbluex.liquidbounce.api.minecraft.item.IItem;
import net.ccbluex.liquidbounce.api.minecraft.item.IItemPotion;
import net.ccbluex.liquidbounce.api.minecraft.item.IItemStack;
import net.ccbluex.liquidbounce.api.minecraft.network.play.client.ICPacketEntityAction;
import net.ccbluex.liquidbounce.api.minecraft.potion.IPotionEffect;
import net.ccbluex.liquidbounce.api.minecraft.potion.PotionType;
import net.ccbluex.liquidbounce.event.EventTarget;
import net.ccbluex.liquidbounce.event.MotionEvent;
import net.ccbluex.liquidbounce.features.module.Module;
import net.ccbluex.liquidbounce.features.module.ModuleCategory;
import net.ccbluex.liquidbounce.features.module.ModuleInfo;
import net.ccbluex.liquidbounce.features.module.modules.combat.AutoPot$WhenMappings;
import net.ccbluex.liquidbounce.injection.backend.WrapperImpl;
import net.ccbluex.liquidbounce.utils.InventoryUtils;
import net.ccbluex.liquidbounce.utils.MinecraftInstance;
import net.ccbluex.liquidbounce.utils.Rotation;
import net.ccbluex.liquidbounce.utils.RotationUtils;
import net.ccbluex.liquidbounce.utils.misc.FallingPlayer;
import net.ccbluex.liquidbounce.utils.misc.RandomUtils;
import net.ccbluex.liquidbounce.utils.timer.MSTimer;
import net.ccbluex.liquidbounce.value.BoolValue;
import net.ccbluex.liquidbounce.value.FloatValue;
import net.ccbluex.liquidbounce.value.IntegerValue;
import net.ccbluex.liquidbounce.value.ListValue;

@ModuleInfo(name="AutoPot", description="Automatically throws healing potions.", category=ModuleCategory.COMBAT)
public final class AutoPot
extends Module {
    private final FloatValue healthValue = new FloatValue("Health", 15.0f, 1.0f, 20.0f);
    private final IntegerValue delayValue = new IntegerValue("Delay", 500, 500, 1000);
    private final BoolValue openInventoryValue = new BoolValue("OpenInv", false);
    private final BoolValue simulateInventory = new BoolValue("SimulateInventory", true);
    private final FloatValue groundDistanceValue = new FloatValue("GroundDistance", 2.0f, 0.0f, 5.0f);
    private final ListValue modeValue = new ListValue("Mode", new String[]{"Normal", "Jump", "Port"}, "Normal");
    private final MSTimer msTimer = new MSTimer();
    private int potion = -1;

    /*
     * Unable to fully structure code
     */
    @EventTarget
    public final void onMotion(MotionEvent motionEvent) {
        if (!this.msTimer.hasTimePassed(((Number)this.delayValue.get()).intValue()) || MinecraftInstance.mc.getPlayerController().isInCreativeMode()) {
            return;
        }
        v0 = MinecraftInstance.mc.getThePlayer();
        if (v0 == null) {
            return;
        }
        thePlayer = v0;
        switch (AutoPot$WhenMappings.$EnumSwitchMapping$0[motionEvent.getEventState().ordinal()]) {
            case 1: {
                potionInHotbar = this.findPotion(36, 45);
                if (thePlayer.getHealth() <= ((Number)this.healthValue.get()).floatValue() && potionInHotbar != -1) {
                    if (thePlayer.getOnGround()) {
                        var4_5 = (String)this.modeValue.get();
                        var5_8 = false;
                        v1 = var4_5;
                        if (v1 == null) {
                            throw new TypeCastException("null cannot be cast to non-null type java.lang.String");
                        }
                        var4_5 = v1.toLowerCase();
                        switch (var4_5.hashCode()) {
                            case 3446913: {
                                if (!var4_5.equals("port")) ** break;
                                break;
                            }
                            case 3273774: {
                                if (!var4_5.equals("jump")) ** break;
                                thePlayer.jump();
                                ** break;
                            }
                        }
                        thePlayer.moveEntity(0.0, 0.42, 0.0);
                    }
lbl27:
                    // 7 sources

                    fallingPlayer = new FallingPlayer(thePlayer.getPosX(), thePlayer.getPosY(), thePlayer.getPosZ(), thePlayer.getMotionX(), thePlayer.getMotionY(), thePlayer.getMotionZ(), thePlayer.getRotationYaw(), thePlayer.getMoveStrafing(), thePlayer.getMoveForward());
                    v2 = fallingPlayer.findCollision(20);
                    collisionBlock = v2 != null ? v2.getPos() : null;
                    v3 = thePlayer.getPosY();
                    v4 = collisionBlock;
                    v5 = v4 != null ? v4.getY() : 0;
                    if (v3 - (double)v5 >= ((Number)this.groundDistanceValue.get()).doubleValue()) {
                        return;
                    }
                    this.potion = potionInHotbar;
                    MinecraftInstance.mc.getNetHandler().addToSendQueue(MinecraftInstance.classProvider.createCPacketHeldItemChange(this.potion - 36));
                    if (thePlayer.getRotationPitch() <= 80.0f) {
                        RotationUtils.setTargetRotation(new Rotation(thePlayer.getRotationYaw(), RandomUtils.INSTANCE.nextFloat(80.0f, 90.0f)));
                    }
                    return;
                }
                potionInInventory = this.findPotion(9, 36);
                if (potionInInventory == -1 || !InventoryUtils.hasSpaceHotbar()) break;
                if (((Boolean)this.openInventoryValue.get()).booleanValue() && !MinecraftInstance.classProvider.isGuiInventory(MinecraftInstance.mc.getCurrentScreen())) {
                    return;
                }
                v6 = openInventory = MinecraftInstance.classProvider.isGuiInventory(MinecraftInstance.mc.getCurrentScreen()) == false && (Boolean)this.simulateInventory.get() != false;
                if (openInventory) {
                    var7_12 = MinecraftInstance.mc.getNetHandler();
                    $i$f$createOpenInventoryPacket = false;
                    v7 = WrapperImpl.INSTANCE.getClassProvider();
                    v8 = LiquidBounce.INSTANCE.getWrapper().getMinecraft().getThePlayer();
                    if (v8 == null) {
                        Intrinsics.throwNpe();
                    }
                    var8_15 = v7.createCPacketEntityAction(v8, ICPacketEntityAction.WAction.OPEN_INVENTORY);
                    var7_12.addToSendQueue(var8_15);
                }
                MinecraftInstance.mc.getPlayerController().windowClick(0, potionInInventory, 0, 1, thePlayer);
                if (openInventory) {
                    MinecraftInstance.mc.getNetHandler().addToSendQueue(MinecraftInstance.classProvider.createCPacketCloseWindow());
                }
                this.msTimer.reset();
                break;
            }
            case 2: {
                if (this.potion < 0 || !(RotationUtils.serverRotation.getPitch() >= 75.0f)) break;
                itemStack = thePlayer.getInventory().getStackInSlot(this.potion);
                if (itemStack != null) {
                    potionInInventory = WEnumHand.MAIN_HAND;
                    var7_13 = MinecraftInstance.mc.getNetHandler();
                    $i$f$createUseItemPacket = false;
                    var8_16 = WrapperImpl.INSTANCE.getClassProvider().createCPacketTryUseItem((WEnumHand)hand$iv);
                    var7_13.addToSendQueue(var8_16);
                    MinecraftInstance.mc.getNetHandler().addToSendQueue(MinecraftInstance.classProvider.createCPacketHeldItemChange(thePlayer.getInventory().getCurrentItem()));
                    this.msTimer.reset();
                }
                this.potion = -1;
            }
        }
    }

    /*
     * WARNING - void declaration
     */
    private final int findPotion(int startSlot, int endSlot) {
        IEntityPlayerSP iEntityPlayerSP = MinecraftInstance.mc.getThePlayer();
        if (iEntityPlayerSP == null) {
            Intrinsics.throwNpe();
        }
        IEntityPlayerSP thePlayer = iEntityPlayerSP;
        int n = startSlot;
        int n2 = endSlot;
        while (n < n2) {
            void i;
            IItemStack stack = thePlayer.getInventoryContainer().getSlot((int)i).getStack();
            if (stack != null && MinecraftInstance.classProvider.isItemPotion(stack.getItem()) && stack.isSplash()) {
                IItem iItem = stack.getItem();
                if (iItem == null) {
                    Intrinsics.throwNpe();
                }
                IItemPotion itemPotion = iItem.asItemPotion();
                for (IPotionEffect potionEffect : itemPotion.getEffects(stack)) {
                    if (potionEffect.getPotionID() != MinecraftInstance.classProvider.getPotionEnum(PotionType.HEAL).getId()) continue;
                    return (int)i;
                }
                if (!thePlayer.isPotionActive(MinecraftInstance.classProvider.getPotionEnum(PotionType.REGENERATION))) {
                    for (IPotionEffect potionEffect : itemPotion.getEffects(stack)) {
                        if (potionEffect.getPotionID() != MinecraftInstance.classProvider.getPotionEnum(PotionType.REGENERATION).getId()) continue;
                        return (int)i;
                    }
                }
            }
            ++i;
        }
        return -1;
    }

    @Override
    public String getTag() {
        return String.valueOf(((Number)this.healthValue.get()).floatValue());
    }
}

File: sources/LiquidSense Dev/net/ccbluex/liquidbounce/features/module/modules/combat/AutoPot.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.combat

import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.event.EventState.POST
import net.ccbluex.liquidbounce.event.EventState.PRE
import net.ccbluex.liquidbounce.event.EventTarget
import net.ccbluex.liquidbounce.event.MotionEvent
import net.ccbluex.liquidbounce.features.module.Module
import net.ccbluex.liquidbounce.features.module.ModuleCategory
import net.ccbluex.liquidbounce.features.module.ModuleInfo
import net.ccbluex.liquidbounce.utils.*
import net.ccbluex.liquidbounce.utils.misc.FallingPlayer
import net.ccbluex.liquidbounce.utils.misc.RandomUtils
import net.ccbluex.liquidbounce.utils.timer.MSTimer
import net.ccbluex.liquidbounce.value.BoolValue
import net.ccbluex.liquidbounce.value.FloatValue
import net.ccbluex.liquidbounce.value.IntegerValue
import net.ccbluex.liquidbounce.value.ListValue
import net.minecraft.client.gui.inventory.GuiInventory
import net.minecraft.item.ItemPotion
import net.minecraft.network.play.client.C08PacketPlayerBlockPlacement
import net.minecraft.network.play.client.C09PacketHeldItemChange
import net.minecraft.network.play.client.C0DPacketCloseWindow
import net.minecraft.network.play.client.C16PacketClientStatus
import net.minecraft.potion.Potion

@ModuleInfo(name = "AutoPot", description = "Automatically throws healing potions.", category = ModuleCategory.COMBAT)
class AutoPot : Module() {

    private val healthValue = FloatValue("Health", 15F, 1F, 20F)
    private val delayValue = IntegerValue("Delay", 500, 500, 1000)

    private val openInventoryValue = BoolValue("OpenInv", false)
    private val simulateInventory = BoolValue("SimulateInventory", true)

    private val groundDistanceValue = FloatValue("GroundDistance", 2F, 0F, 5F)
    private val modeValue = ListValue("Mode", arrayOf("Normal", "Jump", "Port"), "Normal")

    private val msTimer = MSTimer()
    private var potion = -1

    @EventTarget
    fun onMotion(motionEvent: MotionEvent) {
        val killAura = LiquidBounce.moduleManager.getModule(Aura::class.java) as Aura

        if ((mc.thePlayer!!.capabilities.allowFlying && mc.thePlayer!!.capabilities.isFlying) || !msTimer.hasTimePassed(delayValue.get().toLong()) || mc.playerController.isInCreativeMode || (mc.thePlayer!!.openContainer != null && mc.thePlayer!!.openContainer!!.windowId != 0) || killAura.target != null)
            return

        val thePlayer = mc.thePlayer ?: return
        val potionInHotbar = findPotion(36, 45)
        // Inventory Potion -> Hotbar Potion
        val potionInInventory = findPotion(9, 36)


        when (motionEvent.eventState) {
            PRE -> {
                // Hotbar Potion

                if (potionInHotbar != -1 && mc.currentScreen !is GuiInventory) {
                    if (thePlayer.onGround) {
                        when (modeValue.get().toLowerCase()) {
                            "jump" -> thePlayer.jump()
                            "port" -> thePlayer.moveEntity(0.0, 0.42, 0.0)
                        }
                    }

                    potion = potionInHotbar - 36
                    mc.netHandler.addToSendQueue(C09PacketHeldItemChange(potion))

                    if (thePlayer.rotationPitch <= 80F) {

                        RotationUtils.setTargetRotation(Rotation(thePlayer.rotationYaw, RandomUtils.nextFloat(80F, 90F)))
                        if (potion != -1 && RotationUtils.serverRotation.pitch >= 80F) {
                            val itemStack = thePlayer.inventory.getStackInSlot(potion)

                            if (itemStack != null) {
                                mc.netHandler.addToSendQueue(C08PacketPlayerBlockPlacement(itemStack))
                                mc.netHandler.addToSendQueue(C09PacketHeldItemChange(mc.thePlayer.inventory.currentItem))
                            }
                            potion = -1
                            msTimer.reset()
                        }
                    }
                }

                if ((potionInInventory != -1 && potion == -1) && InventoryUtils.hasSpaceHotbar()) {
                    if (openInventoryValue.get() && mc.currentScreen !is GuiInventory)
                        return

                    val openInventory = mc.currentScreen !is GuiInventory && simulateInventory.get()

                    if (openInventory)
                        mc.netHandler.addToSendQueue(
                                C16PacketClientStatus(C16PacketClientStatus.EnumState.OPEN_INVENTORY_ACHIEVEMENT))

                    mc.playerController.windowClick(0, potionInInventory, 0, 1, thePlayer)

                    if (openInventory)
                        mc.netHandler.addToSendQueue(C0DPacketCloseWindow())

                    msTimer.reset()
                }
            }
        }
    }

    private fun findPotion(startSlot: Int, endSlot: Int): Int {
        val thePlayer = mc.thePlayer!!

        for (i in startSlot until endSlot) {
            val stack = thePlayer.inventoryContainer.getSlot(i).stack

            if (stack == null || stack.item !is ItemPotion || !ItemPotion.isSplash(stack.itemDamage))
                continue

            val itemPotion = stack.item as ItemPotion

            if (thePlayer.health <= healthValue.get()) {
                for (potionEffect in itemPotion.getEffects(stack))
                    if (potionEffect.potionID == Potion.heal.id)
                        return i

                if (!thePlayer.isPotionActive(Potion.regeneration.id))
                    for (potionEffect in itemPotion.getEffects(stack))
                        if (potionEffect.potionID == Potion.regeneration.id)
                            return i
            }

            if (!thePlayer.isPotionActive(Potion.moveSpeed.id))
                for (potionEffect in itemPotion.getEffects(stack))
                    if (potionEffect.potionID == Potion.moveSpeed.id)
                        return i
        }

        return -1
    }

    /*  override val tag: String?
        get() = healthValue.get().toString()*/

}